api/src/main/java/io/neba/api/resourcemodels/AnnotatedFieldMapper.java

```diff
@@ -0,0 +1,162 @@
+ *         return MyAnnotation.class;
+ *     }
+ *
+ *     public Collection map(OngoingMapping&lt;Collection, MyAnnotation&gt; ongoingMapping) {
+ *         ...
+ *     }
+ * }
+ * </pre>
+ * <p>
+ * Custom mappers are always invoked <em>after</em> all of NEBA's standard mappings have occurred, but before
+ * the corresponding value was set on the model's field. They may thus make use of the already resolved
+ * value or choose to provide a different one.
+ * </p>
+ *
+ * <p>
+ *     <strong>It is crucial for a {@link AnnotatedFieldMapper} to always return a value that is assignment compatible
+ *     to the {@link OngoingMapping#getFieldType() field type}, i.e. either of the same or of a more specific type.</strong>
+ *     It is insufficient to return a type compatible to the {@link #map(AnnotatedFieldMapper.OngoingMapping) mapping methods}
+ *     return type declaration. This return type only represents the type any returned value must be compatible to.<br />
+ *     For instance, if a mapper is responsible for {@link java.util.Collection}, it must take care to return the field's actual
+ *     collection type, e.g. {@link java.util.List} or {@link java.util.Set}. Otherwise, an exception will arise.
+ * </p>
+ *
+ * <p>
+ *     Implementations <strong>must never</strong> store any contextual data provided by the
+ *     {@link io.neba.api.resourcemodels.AnnotatedFieldMapper.OngoingMapping}
+ *     as this data stems from arbitrary OSGi bundles with independent life cycles.
+ *     Storing any data would result in a class loader / memory leak when these bundles change.
+ * </p>
+ *
+ * @param <FieldType> the super type of the {@link java.lang.reflect.Field#getType() field type}
+ *        of the mapped type.
+ * @param <AnnotationType> the exact type of the annotation this mapper is responsible for.
+
+ * @author Olaf Otto
+ */
+public interface AnnotatedFieldMapper<FieldType, AnnotationType extends Annotation> {
+    /**
+     * Represents the contextual data of a field mapping during a resource to model mapping.
+     *
+     * @author Olaf Otto
+     */
+    public interface OngoingMapping<FieldType, AnnotationType> {
+        /**
+         * @return The currently resolved value of the field,
+         * or <code>null</code> if no value could be resolved for the field.
+         */
+        FieldType getResolvedValue();
+
+        /**
+         * @return The instance of {@link #getAnnotationType() the annotation this mapper is registered for}.
+         *         Never <code>null</code>.
+         */
+        AnnotationType getAnnotation();
+
+        /**
+         * @return The mapped model. At this point, the mapping is still incomplete and
+         *         no post-processors have been invoked on the model. Never <code>null</code>.
+         */
+        Object getModel();
+
+        /**
+         * @return the mapped field. Never <code>null</code>.
+         */
+        Field getField();
```

Comment:
So I assume getField().getValue() != getResolvedValue() at this time? Maybe this should be mentioned here.